src/figuresClasses.ts

```diff
@@ -2,10 +1,82 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error(`Values must be greater than 0`);
+    }
+
+    const sum = this.a + this.b + this.c;
+    const longestSide = Math.max(this.a, this.b, this.c);
+
+    if (longestSide >= sum - longestSide) {
+      throw new Error(`Not a triangle`);
+    }
+  }
+
+  getArea(): number {
+    const semip = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(
+      semip * (semip - this.a) * (semip - this.b) * (semip - this.c),
+    );
+
+    return Math.floor(area * 100) / 100;
+  }
+}
 
-export class Circle implements Figure {}
+export class Circle implements Figure {
+  public shape: Shape = 'circle';
 
-export class Rectangle implements Figure {}
+  constructor(
+    public color: 'red' | 'green' | 'blue',
```

Comment:
Consider using the `Color` type alias instead of repeating the color options. This will ensure consistency and make it easier to update the color options in the future.